pkg/reconciler/v1alpha1/revision/table_test.go

```diff
@@ -1351,10 +1350,10 @@ func TestReconcile(t *testing.T) {
 
 	table.Test(t, MakeFactory(func(listers *Listers, opt reconciler.Options) controller.Reconciler {
 		return &Reconciler{
-			Base:             reconciler.NewBase(opt, controllerAgentName),
-			revisionLister:   listers.GetRevisionLister(),
-			kpaLister:        listers.GetKPALister(),
-			buildLister:      listers.GetBuildLister(),
+			Base:           reconciler.NewBase(opt, controllerAgentName),
+			revisionLister: listers.GetRevisionLister(),
+			kpaLister:      listers.GetKPALister(),
+			//buildLister:      listers.GetBuildLister(), // TODO replace with duck.InformerFactory
```

Comment:
I think we want something that will create a `duck.InformerFactory` around a `GenericLister`, which we would populate from `KResource`s in the input list.

We'd also need to convert these to `unstructured.Unstructured` to seed the dynamic client with them (requires TypeMeta to be specified, which shouldn't be the `KResource` `TypeMeta`).

I can try to sketch something here, if it would be helpful, just let me know (on slack is best).

Reply:
Delete this line?